### PR TITLE
Add workers reconcile for self managed clusters

### PR DESCRIPTION
**Pull request by g-gaston**
## Description of changes
Until now, the new full lifecycle controller completely ignored self managed (management and standalone) clusters. This PR adds partial reconcile support by reacting to changes in the worker part of the spec.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/vsphere/reconciler/reconciler_test.go [hunk 1]
```diff
@@ -68,6 +69,58 @@ func TestReconcilerReconcileSuccess(t *testing.T) {
 	tt.Expect(result).To(Equal(controller.Result{}))
 }
 
+func TestReconcilerReconcileWorkerNodesSuccessSuccess(t *testing.T) {
```
Comment (jonathanmeier5) [inline pkg/providers/vsphere/reconciler/reconciler_test.go:72]: Is `SuccessSuccess` intentional here?

File: controllers/cluster_controller_test_test.go [hunk 2]
```diff
@@ -161,8 +161,12 @@ func TestClusterReconcilerSetBundlesRef(t *testing.T) {
 }
 
 func newRegistryForDummyProviderReconciler() controllers.ProviderClusterReconcilerRegistry {
```
Comment (jonathanmeier5) [inline controllers/cluster_controller_test_test.go:163]: What's the significance of `_test_test` here? are we testing the test file?

Comment (g-gaston) [reply controllers/cluster_controller_test_test.go:163]: Historic reasons: `_test.go` used to be in the `controllers` package and `test_test.go` in `controllers_test`. Now that they are both in `controllers_test`, they can be combined. I just didn't do it as part of this PR to keep the scope down.

**Review by jonathanmeier5 — APPROVED** [review]
This all looks reasonable to me. I defer to others who have more context on this project though.

/lgtm

**Comment by g-gaston** [comment]
Adding lgtm back, I just squashed the commits

File: controllers/cluster_controller_test.go [hunk 3]
```diff
@@ -74,40 +78,61 @@ func newVsphereClusterReconcilerTest(t *testing.T, objs ...runtime.Object) *vsph
...
+	c := fake.NewClientBuilder().WithRuntimeObjects(selfManagedCluster).Build()
 
-	tt := newVsphereClusterReconcilerTest(t, objs...)
+	providerReconciler.EXPECT().ReconcileWorkerNodes(ctx, log, sameName(selfManagedCluster))
 
-	req := clusterRequest(cluster)
+	r := controllers.NewClusterReconciler(c, log, registry)
+	result, err := r.Reconcile(ctx, clusterRequest(selfManagedCluster))
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(result).To(Equal(ctrl.Result{}))
```
Comment (taneyland) [inline controllers/cluster_controller_test.go:107]: Can we add a test with assertions to show that CP was not reconciled & workers were updated?

**Pull request merged**